ncolony/ctllib.py

```diff
@@ -77,7 +77,7 @@ def add(places, name, cmd, args, env=None, uid=None, gid=None, extras=None):
         details['gid'] = gid
     if extras is not None:
         details.update(extras)
-    content = json.dumps(details)
+    content = json.dumps(details).encode('utf-8')
```

Comment:
This happens throughout the code.  I suggest wrapping encoding and decoding JSON to and from bytes in a pair of functions.